manta-crypto/src/merkle_tree/tree.rs

```diff
@@ -388,6 +390,12 @@ pub enum PathError {
         /// Length of the tree
         length: usize,
     },
+
+    /// Given index is smaller than the starting leaf index
+    IndexTooSmall {
+        /// Starting leaf index of the (partial) tree
+        starting_index: usize,
+    },
```

Comment:
These errors are supposed to be generic for all tree implementations, nothing specific to partial trees. Whenever the index is too small, `MissingPath` should be used instead.